### DISCUSSION TITLE
Receiving 404 against private gh repository and different org resolving github-tags

### DISCUSSION THREAD

**Discussion by jeskosda** [renovatebot/renovate#10605] [Request Help]
Hey guys, I have some problems when it comes to authenticating with running Github App Renovate in ORG-A accessing private Repository in ORG-B using the specific hostRule  (also tried out only using hostType=github-tags):

```
"hostRules": [
    {
      "matchHost": "github.com",
      "encrypted": {
        "token": "ENCRYPTED_TOKEN"
      }
    }
]

```

The PAT is from a bot user having access to both github orgs.
When checking the Rate-Limit and the request with that token with the following pattern locally, there is no problem:
`curl -X GET -H "Authorization: token  TOKEN"  "https://api.github.com/repos/ORG-B/REPO/tags?per_page=100"
`

```
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/ORG-B/REPO/tags?per_page=100"
}
DEBUG: Datasource 404
{
  "datasource": "github-tags",
  "lookupName": "ORG-B/REPO",
  "url": "https://api.github.com/repos/ORG-B/REPO/tags?per_page=100"
}
DEBUG: Failed to look up dependency ORG-B/REPO (ORG-B/REPOl)(packageFile="bases/some-base/kustomization.yaml", dependency="ORG-B/REPO")
```

The funny thing is that this seems to be flaky because it has worked twice since two days:
`DEBUG: Fetching changelog: https://github.com/ORG-B/REPO (0.14.1 -> 0.14.3)` and the PR was created, but after the next job it was autoclosed again.

Do you have any thoughts about that? Is something miss-configured?

**Comment by rarkins** [comment]
Please don't override GitHub.com so widely if you're using the app - it could cause chaos. Try matchHost of `https://github.com/orgb/` in case that works. If not then it likely needs a code review to understand if we can get that to work.

**Reply by jeskosda** [reply]
Thanks for the fast response!

Also tried it out with same result: 
```
DEBUG: Adding token authentication for https://github.com/ORG-B/ to hostRules
DEBUG: Adding token authentication for https://api.github.com/repos/ORG-B/ to hostRules
…
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/ORG-B/REPO/tags?per_page=100"
}
DEBUG: Datasource 404
{
  "datasource": "github-tags",
  "lookupName": "ORG-B/REPO",
  "url": "https://api.github.com/repos/ORG-B/REPO/tags?per_page=100"
}
```

If it could cause chaos when using the Github App platform it may be applicable to add also some hint here:
https://docs.renovatebot.com/presets-default/#githubcomtokenltarg0gt ?

**Reply by rarkins** [reply]
Good catch re: `https://api.github.com/repos/ORG-B/`. I think the `https://github.com/ORG-B/` one I suggested is unlikely to be needed.

Next step is you'll need to set up a reproduction of this in a public repo which can be forked and debugged. The "ORG-B" doesn't even need to exist, and the additional token can be dummy - what we'd need to verify is which token is being sent with that request.

**Reply by jeskosda** [reply]
Provided the following repo: https://github.com/jeskosda/renovate-reprod  Hope that helps!

**Reply by jeskosda** [reply]
Hey rarkins! Shall I create an issue for that? Or is the discussion still in the right place?

**Reply by rarkins** [reply]
@jeskosda Please create an issue if you haven't already.

**Reply by jeskosda** [reply]
Already created! https://github.com/renovatebot/renovate/issues/10737
